### PR TITLE
CI(pytest): Add test dependencies for API validation

### PR DESCRIPTION
**Pull request by titom73**
## Change Summary

Avoid running test if Authentication is not set correctly:

- Skip auth if user_token is not set
- Skip if authentication failed.

## Related Issue(s)

N/A

## Component(s) name

pytest

## Proposed changes

Set `test_cvp_connection` as dependency for all tests based on API

```python
@pytest.mark.usefixtures("CvDeviceTools_Manager")
class TestCvDeviceToolsGeneric():

    @pytest.mark.dependency(name='authentication')
    @pytest.mark.skipif(user_token == 'unset_token', reason="Token is not set correctly")
    def test_cvp_connection(self):
        assert True
        logging.info("Connected to CVP")

    @pytest.mark.dependency(depends=["authentication"], scope='class')
    @pytest.mark.api
    def test_search_by_getter_setter(self):
        self.inventory.search_by = FIELD_SYSMAC
        assert self.inventory.search_by == FIELD_SYSMAC
        logging.info(
            "Setter & Getter for search_by using {} is valid".format(FIELD_SYSMAC))
        self.inventory.search_by = FIELD_FQDN
        assert self.inventory.search_by == FIELD_FQDN
        logging.info(
            "Setter & Getter for search_by using {} is valid".format(FIELD_FQDN))
```

## How to test

TBD

### User Checklist

- [x] Finalize all system tests
- [x] Commit Makefile change
- [x] Build documentation
- [x] Activate Pytest in CI

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly. (check the box if not applicable)
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by titom73** [comment]
To be fixed:

```
ERROR: Found 1 pylint issue(s) which need to be resolved:\

ERROR: plugins/module_utils/device_tools.py:1077:36: \
ansible-format-automatic-specification: Format string contains automatic field numbering specification
```

File: tests/system/test_cv_container_tools.py [hunk 1]
```diff
@@ -197,13 +210,16 @@ def test_get_configlet_from_container(self, UNIT_TEST):
 @pytest.mark.filterwarnings("ignore::urllib3.exceptions.InsecureRequestWarning")
 class TestCvContainerToolsTopology():
 
+    @pytest.mark.dependency(name='authentication')
+    @pytest.mark.skipif(user_token == 'unset_token', reason="Token is not set correctly")
```
Comment (ClausHolbechArista) [inline tests/system/test_cv_container_tools.py:214]: Do you need this skipif when you also have the dependency on authentication?

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: tests/README.md [hunk 2]
```diff
@@ -1,47 +1,59 @@
...
+
+```bash
+# User token generated from Cloudvision instance
+export ARISTA_AVD_CV_TOKEN='your-token-from-your-cv-instance'
 
-- Username
-- Password
-- Server
+# Cloudvision address
+export ARISTA_AVD_CV_SERVER='https://lab.cv.io/'
```
Comment (guillaumeVilar) [inline tests/README.md]: I would change the example URL from the README as that would suggest that we need to add the `https://` part at the beginning (authentication with `https://` at the beginning does not work). Maybe just use `www.arista.io` as an example?

**Review by guillaumeVilar — COMMENTED** [review]
Despite my authentication being set, the tests in the following files are always marked as skipped: 
* system/test_cv_device_tools_fqdn.py
* system/test_cv_device_tools_generic.py
* system/test_cv_device_tools_serial.py
* system/test_cv_device_tools.py
The following tests are running as expected: 
* system/test_cv_container_tools.py
* unit/  


My authentication is set as mentioned in the README: 
```
echo $ARISTA_AVD_CV_TOKEN
<MY_CVAAS_TOKEN>

echo $ARISTA_AVD_CV_SERVER
www.mycvpserver.arista.io
```

File: tests/README.md [hunk 3]
```diff
@@ -1,47 +1,59 @@
 # Ansible CVP Collection testing
 
-## Configure Pytest
+## Availabale tests
```
Comment (guillaumeVilar) [inline tests/README.md]: Typo here

File: tests/README.md [hunk 4]
```diff
@@ -1,47 +1,59 @@
...
 ```bash
-$ make config.py
-Generate config.py for Cloudvision auth
----
-Update unit/config.py with your credentials
+make test TESTS=system/
 ```
 
-## Run tests
+- `TAG`: Select pytest tag to run. By defaul, run `api` and `generic`
```
Comment (guillaumeVilar) [inline tests/README.md]: typo: defaul --> default

File: tests/README.md [hunk 5]
```diff
@@ -1,47 +1,59 @@
...
 
-A [Makefile](Makefile) is available to provide commands:
 
-- `test`: Run pytest to execute all tests with `generic` and `api` flags.
-- `test-all`: Run pytest to execute all tags.
-- `repor`: Run pytest to execute all tests with `generic` and `api` flags + HTML reports (pytest + codecoverage).
-- `report-all`: Run pytest to execute all tags + HTML reports (pytest + codecoverage).
+## Tests Results
+
+Results are print to your screen and also saved in reports:
```
Comment (guillaumeVilar) [inline tests/README.md]: typo: print --> printed

**Review by guillaumeVilar — COMMENTED** [review]
Some typos found in the README

File: tests/README.md (hunk 3 above)
Comment (titom73) [reply tests/README.md]: Fixed

File: tests/README.md (hunk 5 above)
Comment (titom73) [reply tests/README.md]: Fixed

File: tests/README.md (hunk 4 above)
Comment (titom73) [reply tests/README.md]: Fixed

**Review by guillaumeVilar — APPROVED** [review]
LGTM, README improved and all tests are running now with `make test`!

**Review by manuwelakanade — COMMENTED** [review]
LGTM

**Comment by KonikaChaurasiya-GSLab** [comment]
reviewed

**Review by KonikaChaurasiya-GSLab — COMMENTED** [review]
reviewed

**Pull request merged**